azavea/ism-watershed-wellness-snapshot
# Initial fetch and save sensor data

## Overview

The app markers will change their display depending on sensor data read regularly from real USGS sensors/data feeds. This app puts into place the initial work connecting the sensors to the app. Certainly more work will follow in terms of error handling, using back up data sources, persisting data, etc.

The requests to the 3 sensors are batch requested and handled at once. There's notably no error handling yet.

Sensors set to poll every 10 minutes, subj to change.

Connects #20

### Demo

![screen shot 2019-02-12 at 5 00 57 pm](https://user-images.githubusercontent.com/10568752/52733471-cfd2b700-2f90-11e9-91af-af254d156407.png)

Sensors polling at 1 minute for testing:
![screen shot 2019-02-12 at 2 00 28 pm](https://user-images.githubusercontent.com/10568752/52733502-e2e58700-2f90-11e9-9b9c-5d2fe5eb34bd.png)

## Notes

I decided to try out `async/await`. I don't know that the implementation is more readable or effective than the standard request/then/catch promise workflow. Let me know your thoughts!

Also, I added changes to the Vagrantfile that helped my VM stop freezing -- it's the same set of change I've been applying to other projects. Matt noted this same change that helps me, to mount files via `nfs`, broke the VM on his linux machine so hopefully this change should work for linux machines too -- untested.

## Testing Instructions
`vagrant reload`
`./scripts/update`
`./scripts/server`
Upon starting or refreshing the app the requests should fire
